src/brevitas/graph/quantize_impl.py

```diff
@@ -0,0 +1,423 @@
+        elif inp_node.op == 'call_function' and inp_node.target in [
+                torch.flatten, torch.reshape, torch.transpose, operator.getitem,
+                operator.__getitem__]:
+            recursive_input_handler(
+                model,
+                inp_node,
+                shared_quant_identity_name,
+                shared_quant_identity,
+                rewriters,
+                quant_identity_map,
+                align_input_quant_fn,
+                align_sign)
+        elif inp_node.op == 'call_function' and inp_node.target is CAT:
+            recursive_input_handler(
+                model,
+                inp_node,
+                shared_quant_identity_name,
+                shared_quant_identity,
+                rewriters,
+                quant_identity_map,
+                align_input_quant_fn,
+                align_sign=True)
+        elif inp_node.op == 'call_method' and inp_node.target in [
+                'view', 'reshape', 'flatten', 'transpose']:
+            recursive_input_handler(
+                model,
+                inp_node,
+                shared_quant_identity_name,
+                shared_quant_identity,
+                rewriters,
+                quant_identity_map,
+                align_input_quant_fn,
+                align_sign)
+        else:
+            rewriters.append(InsertModuleCallAfter(shared_quant_identity_name, inp_node))
+
+
+def _get_quant_module(model, node, quant_identity_map, quant_act_map, unsigned_act_tuple):
+    """
+    Generate a QuantIdentity node that will be used for requantization step around a node.
+    If all inputs to that node are unsigned, the QuantIdentity will also be unsigned, otherwise it
+    is signed.
+    """
+    if are_inputs_unsigned(model, node, [], quant_act_map, unsigned_act_tuple):
+        quant_module_class, quant_module_kwargs = quant_identity_map['unsigned']
+    else:
+        quant_module_class, quant_module_kwargs = quant_identity_map['signed']
+    quant_module = quant_module_class(**quant_module_kwargs)
+    quant_module_name = node.name + '_quant'
+    model.add_module(quant_module_name, quant_module)
+    return quant_module, quant_module_name
+
+
+def residual_handler(
+        model, quant_identity_map, quant_act_map, unsigned_act_tuple, align_input_quant_fn):
+
+    def is_converged(model):
+
+        for node in model.graph.nodes:
+            if (node.op == 'call_function' and node.target in ADD_FNS + [CAT] or
+                    node.op == 'call_method' and node.target in ADD_METHODS):
+                rewriters = []
+                # If the op is CAT, check that inputs have same sign, and in recursive_input_handler
+                # force that the sign is aligned
+                same_sign = node.target is CAT
+
+                # If input to the CAT or ADD node are quantized and aligned correctly, continue to
+                # the next node
+                if are_inputs_quantized_and_aligned(model,
```

Comment:
is the formatting ok here?

Reply:
Maybe the function call inside an _if_ is different than a normal function call